File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/RotationVectorSensorService.java
// Copyright 2015-present 650 Industries. All rights reserved.

package expo.modules.sensors.services;

import android.content.Context;
import android.hardware.Sensor;

import java.util.Collections;
import java.util.List;

import org.unimodules.core.interfaces.InternalModule;

import expo.modules.interfaces.sensors.services.RotationVectorSensorServiceInterface;

public class RotationVectorSensorService extends SubscribableSensorService implements InternalModule, RotationVectorSensorServiceInterface {
  public RotationVectorSensorService(Context context) {
    super(context);
  }

  @Override
  int getSensorType() {
    return Sensor.TYPE_ROTATION_VECTOR;
  }

  @Override
  public List<Class> getExportedInterfaces() {
    return Collections.<Class>singletonList(RotationVectorSensorServiceInterface.class);
  }
}

File: packages/expo-sensors/android/src/main/java/expo/modules/sensors/services/RotationVectorSensorService.kt
// Copyright 2015-present 650 Industries. All rights reserved.
package expo.modules.sensors.services

import android.content.Context
import android.hardware.Sensor
import expo.modules.interfaces.sensors.services.RotationVectorSensorServiceInterface
import expo.modules.core.interfaces.InternalModule

class RotationVectorSensorService(context: Context?) : SubscribableSensorService(context), InternalModule, RotationVectorSensorServiceInterface {
  override val sensorType: Int = Sensor.TYPE_ROTATION_VECTOR

  override fun getExportedInterfaces(): List<Class<*>> {
    return listOf<Class<*>>(RotationVectorSensorServiceInterface::class.java)
  }
}
